src/classes/RD2_ValidationService_TEST.cls

```diff
@@ -200,6 +212,208 @@ private with sharing class RD2_ValidationService_TEST {
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsBlankForMonthlyInstallmentPeriod() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriodMonthly()
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+
+        System.assert(errMessage.contains(System.Label.RD2_DayOfMonthIsRequiredForMonthlyInstallment));
+
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month is blank
+    */
+    @isTest
+    private static void shouldNotCreateRDWhenInstallmentPeriodIsNotValid() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriod('Occasionally')
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        String expectedMessage = String.format(
+            System.Label.RD2_InstallmentPeriodMustBeValid,
+            new String[]{'Occasionally' }
+        );
+        System.assert(errMessage.contains(expectedMessage));
+
+    }
+
+    /***
+    * @description Verifies if it displays error message when installment period
+        is monthly and day of month value is not valid
+    */
+    @isTest
+    private static void shouldFailWhenDayOfMonthIsNotValidForMonthlyInstallments() {
+        String errMessage = '';
+        RD2_EnablementService_TEST.setRecurringDonations2Enabled();
+
+        Contact con = UTIL_UnitTestData_TEST.getContact();
+        insert con;
+
+        npe03__Recurring_Donation__c rd = new TEST_RecurringDonationBuilder()
+            .withContact(con.Id)
+            .withAmount(10)
+            .withInstallmentPeriodMonthly()
+            .withDayOfMonth('31')
+            .build();
+
+        try {
+            insert rd;
+        } catch (Exception e) {
+            errMessage = e.getMessage();
+        }
+        String expectedMessage = String.format(
+            System.Label.RD2_DayOfMonthMustBeValid,
+            new String[]{'31' }
+        );
+        System.assert(errMessage.contains(expectedMessage));
+
+    }
+
+    /***
+    * @description Verifies if it creates RDs when installment period
+        is monthly and day of month value is valid
+    */
+    @isTest
+    private static void shouldCreateRDWhenDayOfMonthIsValid() {
```

Comment:
Let's create a unit test verifying an RD with "Day of Month" = LAST_DAY can be created.